build-support/protoc_version.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+# Extract the version of protoc that we are using from our
+# .protoc-version file.
+#
+# This can be used as a source of truth any place that need access to
+# this information (e.g., build arguments for container images to
+# ensure base images are in-sync with our Rust version)
+#
+# It can safely be invoked from anywhere in the repository, as it
+# resolves the .protoc-version file from the repository root.
+set -euo pipefail
+
+repo_root="$(git rev-parse --show-toplevel 2> /dev/null)"
+readonly repo_root
+cat "${repo_root}/.protoc-version"
```

Comment:
I like the centralization of this logic, but in this particular case, I think that as long as we have Python code that is being generated from the Protobuf definitions, we should set the version in `pants.toml`, rather than `.protoc-version`.

See https://www.pantsbuild.org/docs/reference-protoc for more details.

We can still use this script to ensure that the value can get into our Docker images, but it would need to do some TOML-munging. I'm not aware of any shell utilities that deal with TOML, but we could potentially get away with some `grep`, or use a tiny Python script. I'm currently chatting with the Pants folks to see if there's some kind of introspection ability in Pants already.